Makefile

```diff
@@ -29,16 +33,15 @@ all: build
 
 build: $(OBJECTS)
 
-pre:
+$(OBJDIR):
 	@mkdir -p $(OBJDIR)
 
-# Compile each of the binaries
-$(CMD_OBJECTS): pre
-	@echo [go] bin/$@
-	@go build -o ./bin/$@ -ldflags \
+$(CMD_BINS): build_cmds
+
+build_cmds: | $(OBJDIR)
```

Comment:
Oh and the other reason to let Make run `go install` each time is that I think `go` is better at knowing which files a given build depends on than Make would be. We could do a `find *.go`, but there might be C files too, yadda yadda yadda.